matRad_calcPhotonDose.m

```diff
@@ -91,25 +129,35 @@
 
 % Allocate space for dij.physicalDose sparse matrix
 for i = 1:dij.numOfScenarios
-    dij.physicalDose{i} = spalloc(prod(ct.cubeDim),numOfColumnsDij,1);
+    dij.physicalDose{i} = spalloc(dij.doseGrid.numOfVoxels,numOfColumnsDij,1);
 end
 
 % Allocate memory for dose_temp cell array
 doseTmpContainer     = cell(numOfBixelsContainer,dij.numOfScenarios);
 
 % take only voxels inside patient
-V = [cst{:,4}];
-V = unique(vertcat(V{:}));
+VctGrid = [cst{:,4}];
+VctGrid = unique(vertcat(VctGrid{:}));
 
 % ignore densities outside of contours
-eraseCtDensMask = ones(dij.numOfVoxels,1);
-eraseCtDensMask(V) = 0;
+eraseCtDensMask = ones(prod(ct.cubeDim),1);
+eraseCtDensMask(VctGrid) = 0;
 for i = 1:ct.numOfCtScen
     ct.cube{i}(eraseCtDensMask == 1) = 0;
 end
 
 % Convert CT subscripts to linear indices.
-[yCoordsV_vox, xCoordsV_vox, zCoordsV_vox] = ind2sub(ct.cubeDim,V);
+[yCoordsV_vox, xCoordsV_vox, zCoordsV_vox] = ind2sub(ct.cubeDim,VctGrid);
+
+% receive linear indices and grid locations from the dose grid
+tmpCube    = zeros(ct.cubeDim);
+tmpCube(VctGrid) = 1;
+% interpolate cube
+VdoseGrid = find(interp3(dij.ctGrid.y,  dij.ctGrid.x,   dij.ctGrid.z,tmpCube, ...
```

Comment:
I think the second vector needs to be transposed. I am preparing a pull request providing one function for down/upsampling cst = matRad_resizeCstToDoseResolution(cst,dij);

Reply:
why not have the function in a more modular design with the following interface

matRad_resizeCstToGrid(cst,oldGrid,newGrid)

?